develop-guide/writeparser/writeparser.rst

```diff
@@ -1,47 +1,558 @@
+        output flow-control is unsupported, input flow-control is unsupported
+        ARP type: ARPA, ARP Timeout 04:00:00
+        Last input 00:00:00, output 00:00:00, output hang never
+        Last clearing of "show interface" counters never
+        ...
+    ...
+
+
+#. Check the indentation in the output. The indentation tells you about the parent-child relationship of the keys.
+
+   .. note:: Remember to use the indentation (parent-child relationships) to ensure that values don't overwrite other values at the same level. In the following example, the keys for :monospace:`interface_name` are indented so that the :monospace:`mac_address` value for Interface1 won't be overwritten by the :monospace:`mac_address` value for Interface2.
+
+   Your schema might begin with the following lines:
+
+   .. code-block:: python
+
+        'interfaces': {
+            'interface_name': {
+                'oper_status': str,
+                'line_protocol': str,
+                'hardware': str,
+                'mac_address': str,
+                ...
+            ...
+
+
+#. You can also check for ways to group the data based on counters, input and output, as well as other statistics. For the following output:
+
+   .. code-block:: python
+
+        4243 packets input, 361948 bytes, 0 no buffer
+        Received 0 broadcasts (0 IP multicasts)
+        0 runts, 0 giants, 0 throttles
+        0 input errors, 0 CRC, 0 frame, 0 overrun, 0 ignored
+        0 watchdog, 0 multicast, 0 pause input
+        3616 packets output, 1637917 bytes, 0 underruns
+        0 output errors, 0 collisions, 0 interface resets
+        0 unknown protocol drops
+        0 babbles, 0 late collision, 0 deferred
+        0 lost carrier, 0 no carrier, 0 pause output
+        0 output buffer failures, 0 output buffers swapped out
+
+   Your schema could be::
+
+    'counters': {  # categorize the value as 'counters'
+        'input': {   # categorize the 'input' related values
+            'packets': int,
+            'bytes': int,
+
+        },
+        'output': {
+            'packets': int,
+            'bytes': int,
+        }
+
+#. To see the output with a more readable structure, you can use Python's "pretty-print" module:
+
+   .. code-block:: python
+
+    output = dev.parse('show interfaces')
+    import pprint
+    pprint.pprint(output)
+
+   *Result*: The following snippet shows the output formatted so that it's easier to read::
+
+    {'GigabitEthernet1': {'arp_timeout': '04:00:00',
+                      'arp_type': 'arpa',
+                      'auto_negotiate': True,
+                      'bandwidth': 1000000,
+                      'counters': {'in_broadcast_pkts': 0,
+                                   'in_crc_errors': 0,
+                                   'in_errors': 0,
+                                   'in_frame': 0,
+
+   .. note:: You can use ``dev.parse`` even without a schema, because ``genie.libs.parse`` can parse at least some of the output.

```

Comment:
I dont understand your comment here

Reply:
Not my comment. Was part of the guide already. Forgot to remove it after confirming with you that schema is mandatory

Reply:
Fixed now